zerver/lib/actions.py

```diff
@@ -4566,14 +4566,17 @@ def do_update_message(user_profile: UserProfile, message: Message,
 
         new_stream_sub_ids = [user.user_profile_id for user in subs_to_new_stream]
 
-        # Get guest users who aren't subscribed to the new_stream.
-        guest_subs_losing_access = [
+        # Get users who aren't subscribed to the new_stream.
+        subs_losing_access = [
             sub for sub in subscribers
+            if sub.user_profile_id not in new_stream_sub_ids
+        ]
+        guest_subs_losing_access = [
+            sub for sub in subs_losing_access
             if sub.user_profile.is_guest
-            and sub.user_profile_id not in new_stream_sub_ids
         ]
         ums = ums.exclude(user_profile_id__in=[
-            sub.user_profile_id for sub in guest_subs_losing_access])
+            sub.user_profile_id for sub in subs_losing_access])
```

Comment:
Thinking about this from the standpoint of extending this to handle private streams, I think we want new variable names:
* `subs_losing_access` is what we should call the current-in-this-pr `guest_subs_losing_access` -- and that should be defined like this:
```
        subs_losing_usermessages = [
              sub for sub in subscribers if sub.user_profile_id not in new_stream_sub_ids
        ]
        subs_losing_access = [
￼            sub for sub in subs_losing_usermessages
￼            if (sub.user_profile.is_guest or stream.is_private)
￼            and sub.user_profile_id not in new_stream_sub_ids
￼        ]
```